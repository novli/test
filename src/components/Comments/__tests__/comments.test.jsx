import { CommentsComponent } from '../';

const setup = () => {
  const props = {
    page: 1,
    commentsList: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    loading: false,
    loadFirstPage: jest.fn(),
    loadNextPage: jest.fn(),
  };

  const wrapper = shallow(<CommentsComponent {...props} />);

  return {
    props,
    wrapper,
    button: wrapper.find('.button'),
  };
};

describe('components', () => {
  describe('Comments', () => {
    test('component is rendered', () => {
      const { wrapper } = setup();

      expect(wrapper).toMatchSnapshot();
    });

    test('click calls loadNextPage', () => {
      const { props: { loadNextPage }, button } = setup();

      button.simulate('click');

      expect(loadNextPage).toBeCalled();
    });
  });
});
