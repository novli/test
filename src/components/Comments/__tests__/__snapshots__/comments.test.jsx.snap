// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Comments component is rendered 1`] = `
<React.Fragment>
  <table>
    <thead>
      <tr>
        <td>
          id
        </td>
        <td>
          name
        </td>
        <td>
          body
        </td>
        <td>
          email
        </td>
      </tr>
    </thead>
    <tbody>
      <Connect(CommentComponent)
        id={1}
        key="1"
      />
      <Connect(CommentComponent)
        id={2}
        key="2"
      />
      <Connect(CommentComponent)
        id={3}
        key="3"
      />
      <Connect(CommentComponent)
        id={4}
        key="4"
      />
      <Connect(CommentComponent)
        id={5}
        key="5"
      />
      <Connect(CommentComponent)
        id={6}
        key="6"
      />
      <Connect(CommentComponent)
        id={7}
        key="7"
      />
      <Connect(CommentComponent)
        id={8}
        key="8"
      />
      <Connect(CommentComponent)
        id={9}
        key="9"
      />
      <Connect(CommentComponent)
        id={10}
        key="10"
      />
    </tbody>
  </table>
  <div>
    <strong>
      Page: 
      2
    </strong>
     
    <i />
  </div>
  <button
    className="button"
    onClick={[Function]}
  >
    Load Next Page
  </button>
</React.Fragment>
`;
